base/MAPL_SwathGridFactory.F90

```diff
@@ -0,0 +1,1259 @@
+      integer, allocatable, intent(out) :: global_start(:)
+      integer, allocatable, intent(out) :: global_count(:)
+      integer, optional, intent(out) :: rc
+
+      _UNUSED_DUMMY(this)
+      _UNUSED_DUMMY(grid)
+      _UNUSED_DUMMY(local_start)
+      _UNUSED_DUMMY(global_start)
+      _UNUSED_DUMMY(global_count)
+
+      _FAIL('unimplemented')
+      _RETURN(_SUCCESS)
+   end subroutine generate_file_corner_bounds
+
+   function generate_file_reference2D(this,fpointer) result(ref)
+      use pFIO
+      type(ArrayReference) :: ref
+      class(SwathGridFactory), intent(inout) :: this
+      real, pointer, intent(in) :: fpointer(:,:)
+      _UNUSED_DUMMY(this)
+      ref = ArrayReference(fpointer)
+   end function generate_file_reference2D
+
+   function generate_file_reference3D(this,fpointer,metaData) result(ref)
+      use pFIO
+      type(ArrayReference) :: ref
+      class(SwathGridFactory), intent(inout) :: this
+      real, pointer, intent(in) :: fpointer(:,:,:)
+      type(FileMetaData), intent(in), optional :: metaData
+      _UNUSED_DUMMY(this)
+      ref = ArrayReference(fpointer)
+   end function generate_file_reference3D
+
+  subroutine get_ncfile_dimension(filename, nlon, nlat, tdim)
+    use netcdf
+    implicit none
+    !
+    character(len=*), intent(in) :: filename
+    integer, intent(out) :: nlat, nlon, tdim
+    integer  :: ncid , iret, dimid
+    !
+    iret = nf90_open(trim(fileName), NF90_NOWRITE, ncid)
+    !
+    iret = nf90_inq_dimid(ncid, "time", dimid)
+    iret = nf90_inquire_dimension(ncid, dimid, len=tdim)
+    !
+    iret = nf90_inq_dimid(ncid, "lon", dimid)
+    iret = nf90_inquire_dimension(ncid, dimid, len=nlon)
+    !
+    iret = nf90_inq_dimid(ncid, "lat", dimid)
+    iret = nf90_inquire_dimension(ncid, dimid, len=nlat)
+    !
+    iret = nf90_close(ncid)
+    !!write(6,*) "nlat, nlon, tdim = ", nlat, nlon, tdim
+  end subroutine get_ncfile_dimension
+
+  
+subroutine get_v2d_netcdf(filename, name, array, Xdim, Ydim)
+  use netcdf
+  implicit none
+  character(len=*), intent(in) :: name, filename
+  integer, intent(in) :: Xdim, Ydim
+  real, dimension(Xdim,Ydim), intent(out) :: array
+  integer :: ncid
+  !
+  integer :: iret, varid
+  real    :: scale_factor, add_offset
+  !
+  call cknc (  nf90_open      (trim(fileName), NF90_NOWRITE, ncid)  )
+  call cknc (  nf90_inq_varid (ncid,  name,  varid) )
+  call cknc (  nf90_get_var   (ncid, varid,  array) )
+  !
+  iret = nf90_get_att(ncid, varid, 'scale_factor', scale_factor)
+  if(iret .eq. 0) array = array * scale_factor
+  !
+  iret = nf90_get_att(ncid, varid, 'add_offset', add_offset)
+  if(iret .eq. 0) array = array + add_offset
+  !
+  iret = nf90_close(ncid)
+  !
+end subroutine get_v2d_netcdf
+
+
+subroutine cknc(status)
+  use netcdf
+  implicit none
+  integer, intent (in) :: status
+  if(status /= nf90_noerr) then
+     write(6, '(2x,a)')  trim(nf90_strerror(status))
+     stop 2
+  end if
```

Comment:
This routine should probably just go.   And macros used above.   I'll attempt to demonstrate there.